V2

papayaah opened the pull request:
Hi @bryanmtl !

here are the changes that I moved out of the glossier code.

papayaah committed:
permit next_renewal_at

papayaah committed:
state machine to handle when a subscription is being renew

papayaah committed:
changes to accommodate the subscription state machine

papayaah committed:
locale settings for the subscription cancellation reasons

papayaah committed:
show cancellation data on the admin subscription page

papayaah committed:
migration to add the renewal dates

papayaah committed:
subscription renewal active job

papayaah committed:
send admin renewal to the subscription renewal job

papayaah committed:
renew api endpoint

papayaah committed:
cancellation reasons and feedback fields

bryanmtl commented:
Thanks @mickeyren. Are there any specs that go along with these changes?

papayaah commented:
@bryanmtl i can imagine writing a spec for the subscription renewal job and checking for the cancellation fields.

Can you grant me admin settings? I like to put this up on travis.org.

papayaah commented:
@bryanmtl there's a bunch of errors coming out of the spec suite, even on the `master` branch but I'm more than halfway through. Just a little setback as I get myself acquainted with [state_machines](https://github.com/state-machines/state_machines)

papayaah committed:
moved all the rest of the possible states into the subscription state machine

papayaah committed:
removing currency, will be depracted on solidus v2

papayaah committed:
use Spree during  a call to locale setting so it is scoped to the spree namespace

papayaah committed:
make subscription factory start on the active state

bryanmtl commented:
Hey @mickeyren - I've enabled Travis on this repo. Let me know if you have access now.

papayaah committed:
travis setting to fire off Travis CI

papayaah committed:
bump up solidus version

papayaah committed:
refactor some specs, by providing some convenient method to pull next_shipment_dates from the subscription

papayaah committed:
remove portrait_small_url, this doesn't belong here and should be in the app using this

Commit:
fix subscription state column translation

Commit:
fix subscription state column translation

papayaah committed:
tabify the admin edit subscription page to be consistent with v1.4

papayaah committed:
make sure adjust sku and failed renewals only show up under the subscription listing page

papayaah committed:
correct way to add menu to the admin sidebar
also added submenu for the subscriptions tab

papayaah committed:
conflict fix
Merge branch 'v1.4' into v2

* v1.4:
  Adjusts how buttons are added to admin UI

papayaah committed:
rename to solidus_subscriptions

papayaah commented:
@bryanmtl getting there. I actually had to clean up a bunch of issues on the admin pages, improving the UI along the way. Now has tabs! 

![image](https://cloud.githubusercontent.com/assets/175045/20330506/5e13b02e-abda-11e6-94f5-12f776886e24.png)

So far I've been fixing failing specs and admin pages issues so I still yet to write the specs for the new code that were introduced in this PR.

bryanmtl commented:
@mickeyren Great work! Thanks for the update.

papayaah committed:
install phantom js before travis runs tests

papayaah committed:
no need for npm install

papayaah committed:
do not unpause if resume is in the future

papayaah committed:
use poltergeist for capybara

papayaah committed:
added capybara screenshot to test
remove duplicate subscription links

papayaah committed:
provide missing en setting that's breaking the spec
also fix state - when resuming at a later period, state should remain at paused

papayaah committed:
using poltergeist as the default driver for capybara seem to be breaking the specs

papayaah committed:
subscription renewal spec

papayaah commented:
bryanmtl specs are green, well sort of. and i've put in a simple spec for the subscription renewal. 

There is this one last failing spec that happens only on Travis. It's failing to reach the Stripe gateway. 

![image](https://cloud.githubusercontent.com/assets/175045/20452049/912f3348-ae3c-11e6-8d4b-132686542464.png)

this is the spec:

```
before(:each) do
    Stripe.api_key = "sk_test_jTiNI1BxjFxBr4TUqdHefc1f"

    user = create(:user)
    setup_subscriptions_for user
    sign_in_as! user

    my_account = MyAccount::Page.new
    @credit_card = my_account.any_subscription.edit.payment_details
  end

  scenario "displays the current payment info" do
    expect(@credit_card.current_payment_info).to_not be_nil
  end

  scenario "can update payment info" do
    @credit_card.number = "4242424242424242"
    @credit_card.name = "John Doe"
    @credit_card.expiry = "06/20"
    @credit_card.code = "123"

    @credit_card.submit

    expect(@credit_card.current_payment_info).to have_text("4242")
    expect(@credit_card.current_payment_info).to have_text("John Doe")
    expect(@credit_card.current_payment_info).to have_text("6/2020")
  end
```

it's failing on the last one. 

cc @aamyot

bryanmtl commented:
@mickeyren Do you need test Stripe credentials to get this passing? Does that spec work for you locally?

papayaah committed:
fix product selection
remove tooltip - non existent anymore.

papayaah committed:
show sku of the variant

papayaah committed:
fix editing of item quantity and deleting

papayaah committed:
clean up and fix a few locale settings

papayaah committed:
rubocop fixes

sherryson committed:
updates adjust sku service to ignore canceled subscriptions and updates the variant id of subscription items

sherryson committed:
Merge branch 'v2' into hotfix/update-adjust-sku

* v2: (37 commits)
  clean up and fix a few locale settings
  fix editing of item quantity and deleting
  show sku of the variant
  fix product selection remove tooltip - non existent anymore.
  subscription renewal spec
  using poltergeist as the default driver for capybara seem to be breaking the specs
  provide missing en setting that's breaking the spec also fix state - when resuming at a later period, state should remain at paused
  added capybara screenshot to test remove duplicate subscription links
  use poltergeist for capybara
  do not unpause if resume is in the future
  no need for npm install
  install phantom js before travis runs tests
  rename to solidus_subscriptions
  correct way to add menu to the admin sidebar also added submenu for the subscriptions tab
  make sure adjust sku and failed renewals only show up under the subscription listing page
  tabify the admin edit subscription page to be consistent with v1.4
  fix subscription state column translation
  fix subscription state column translation
  remove portrait_small_url, this doesn't belong here and should be in the app using this
  refactor some specs, by providing some convenient method to pull next_shipment_dates from the subscription
  ...

Commit:
Merge pull request #15 from glossier/hotfix/update-adjust-sku

fixes adjust sku service for subscriptions

Commit:
Allow a renewing subscription to be cancelled

papayaah commented:
@bryanmtl @aamyot 

would you guys mind looking over the strange error? 

```
$ bundle exec rake test_app
Expected string default value for '--rc'; got false (boolean)
rake aborted!
NoMethodError: undefined method `last_comment' for #<Rake::Application:0x000000019a4bf0>
```

I don't find any trace of `last_comment` anywhere and the spec suite passes just fine for me locally.

papayaah committed:
added last_completed_order to better reflect what the method really pulls
fix the last order in the admin listing page so it correctly shows the actual last order

papayaah committed:
correct syntax for the controller callbacks

papayaah committed:
fix travis script
ensure a working versionr Rakefile

hugobast committed:
Match local version with Travis' version

hugobast committed:
Travis is making http requests to stripe

And this is causing issues with a particular test. In order to avoid
that I reached out to my trusty VCR and threw it at the problematic
spec. I've noticed that other tests are doing HTTP too but I don't want
to do a full pass at this point.

hugobast committed:
Merge pull request #16 from glossier/fix/syntax-on-subscription-controller

correct syntax for the controller callbacks

papayaah committed:
Fix/intl renewal (#17)

* surface the currency used in the order to easily identify where the order came from

* there has been a failure with the renewal for international orders, the shipping rates
are being calculated for US addresses by default, if the currency is not set in the order.

Setting the currency for new orders generated ensures the right shipping methods are pulled
to match the customer's address

* a little comment to understand why the update of email comes after the creation of the order

* specs to ensure the currency and store id is set from the subscription's last completed order

* refactor into a callback, for clarity and less comments

sherryson committed:
Set order currency in variant autocomplete (#18)

* use subscriptions last completed orders currency when adding items to subscription

* moves currency logic to subscription model

papayaah committed:
Fix/creating subscription from order (#19)

* we don't do weeks so let's default to 1 month

* correct syntax to build a subscription from creating a new one from an order

should have also been saving the order's email

* swap HABTM with has_many as HABTM has many limitations

changing the default interval to months as we don't do weeks

* html fixes for the page

* the model representation for the order subscription join table

* changes to the spec to reflect the new interval type

papayaah committed:
Fix/copying addresses (#20)

* need to exclude those fields that are not present in the subscription address table

* support for UK addresses

* the same here - non-existing fields in the subscription address should be excluded

papayaah committed:
filter option and listing stuck renewing subscriptions (#21)

* initial work on having a page to display subscriptions stuck in the renewal process

* added filter by status on the subscriptions listing page

start showing the renewing subscription. these are subscription(state) that were stuck therefore they belong to this list

* decided to just add a filter and add the renewing state in the failure page

* as required by travis when creating the dummy app:
"deface is required to run solidus_auth_devise with solidus versions < 2.5.
Please add deface to your Gemfile"

also time to bump up the gem version. this is now using a state machine
to set the current state of the subscription

papayaah committed:
Feature/admin improvements (#22)

* stop showing cancelled subscriptions and sort by recent subscription renewal

* clearer labels

* this should fix the sequence of orders appearing in the list

* spec to test that filtering by subscription status works

* initial work on the spec for testing the failures listing

* specs to test that only subscriptions that failed renewing are shown, active and cancelled are not
